crates/primitives/src/chain_spec.rs

```diff
@@ -107,61 +114,103 @@ impl ChainSpec {
     }
 
     /// Returns the supported hardforks and their fork block numbers
-    pub fn hardforks(&self) -> &BTreeMap<Hardfork, BlockNumber> {
+    pub fn hardforks(&self) -> &BTreeMap<Hardfork, ForkKind> {
         &self.hardforks
     }
 
     /// Get the first block number of the hardfork.
```

Comment:
Document that this only works for `ForkKind::Block` *or* `ForkKind::TTD` if the merge netsplit block is known beforehand